Umbraco Commerce v15 updates (#6637)
umbraco/UmbracoDocs: 15/umbraco-commerce/release-notes/v15.0.0-rc.md

---
description: >-
  Umbraco Commerce v15.0.0-rc release notes.
---

# Umbraco Commerce v15.0.0-rc

Umbraco Commerce v15.0.0-rc is the initial release of Umbraco Commerce for Umbraco CMS v15. Because of 

## Key Takeaways

* Everything is now [async](#async).
* [Storefront API](#storefront-api) aligned with Management API
* A number of [Umbraco v15 updates](#umbraco-v15-updates).

## Async

> The key focus of this release is the move to a fully asyncronus code base. In order to reduce the maintenance burden, the decision was made to go fully async without maintaining backwards compatibility. This will therfore require code updates to use the new async methods.

Reviewer: ```suggestion
The key focus of this release is the move to a fully asynchronous code base. To reduce the maintenance burden, the decision was made to go fully async without maintaining backward compatibility. This will therefore require code updates to use the new async methods.
```

### Previous behavior
Previously all C# API's were synchronous and thus blocking by nature.

```csharp
// Fluent API example
order.AddProduct(productRef, 1)
  .SetPaymentMethod(paymentMethodId)
  .SetShippingMethod(shippingMethodId);

// Service method examples
orderService.SaveOrder(order);
emailTemplateService.SendEmail(emailTemplateId, order);

// Notification events
public class MyNotification : NotificationEventHandlerBase<OrderFinalizedNotification>
{
  public override void Handle(OrderFinalizedNotification evt)
  {
      // Implement your custom logic here
  }
}
```

### New behavior

> All API's are now asynchronous and thus are suffixed with Async and return a Task<T> result

Reviewer: ```suggestion
All APIs are now asynchronous and thus are suffixed with Async and return a Task<T> result.
```

```csharp
// Fluent API example
await order.AddProductAsync(productRef, 1)
  .SetPaymentMethodAsync(paymentMethodId)
  .SetShippingMethodAsync(shippingMethodId);

// Service method examples
await orderService.SaveOrderAsync(order);
await emailTemplateService.SendEmailAsync(emailTemplateId, order);

// Notification events
public class MyNotification : NotificationEventHandlerBase<OrderFinalizedNotification>
{
  public override Task HandleAsync(OrderFinalizedNotification evt, CancelationToken cancelationToken)
  {
      // Implement your custom logic here
  }
}
```

## Storefront API

> When implementing the Management API in v14 we learnt a lot about API structure, and also found some common models between the Management API and the Storefront API. To aid with maintenance, and align approaches, the Storefront API was updated to similar patterns.

Reviewer: ```suggestion
Implementing the Management API in v14 provided valuable insights into API structure. Common models were identified between the Management API and the Storefront API. To aid with maintenance and align approaches, the Storefront API was updated to reflect similar patterns.
```

> For the most part the API should stay the same, the key different being changes in operation IDs allowing the Storefront API to be used with client generators.

Reviewer: ```suggestion
The API will largely remain the same, with the main change being updated operation IDs for compatibility with client generators.
```

## Umbraco v15 updates

> As well as the async works, Umbraco Commerce v15 has also been updated to depend on Umbraco v15 which consists of the following updates.

Reviewer: ```suggestion
In addition to the asynchronous work, Umbraco Commerce v15 has been updated to depend on Umbraco v15, which includes the following updates:
```

* Runs against .NET 9
* Variants property editor supports content variants

## What to Test and How to Give Feedback

> We would welcome feedback on any installation / upgrade issues along with bugs found in any of the completed sections as detailed above.

Reviewer: ```suggestion
We welcome any feedback on installation or upgrade issues, as well as any bugs found in the sections mentioned above.
```
